packages/medusa/src/api/routes/admin/sales-channels/add-product-batch.ts

```diff
@@ -32,12 +33,18 @@ export default async (req: Request, res: Response): Promise<void> => {
     "salesChannelService"
   )
 
-  const validatedBody =
-    req.validatedBody as AdminPostSalesChannelsChannelProductsBatchReq
-  const salesChannel = await salesChannelService.addProducts(
-    id,
-    validatedBody.product_ids.map((p) => p.id)
-  )
+  const manager: EntityManager = req.scope.resolve("manager")
+  const salesChannel = await manager.transaction(async (transactionManager) => {
+    const validatedBody =
+      req.validatedBody as AdminPostSalesChannelsChannelProductsBatchReq
```

Comment:
nit(non-blocking): consistency, should this be the first operation of the method and not in the transaction? (as you've done in collections in the other pr https://github.com/medusajs/medusa/pull/1853

(same for all sc endpoints)

Reply:
done